### ISSUE TITLE
How to drop log_event when the async queue is full

### ISSUE THREAD

**Issue by nonly** [log4cplus/log4cplus#271]
hi, some questiens about asyncappender. 

I want to drop log_event when the async queue is full. but I find log4cplus perhaps cannot provide the drop option.

 I find the put_event() can block on internal semaphore, so I should modify the code?

**Comment by wilx** [comment]
Yes. It blocks if the queue is full. You would have to modify the code to get it to drop the logging event.
